test-common/src/main/java/org/triplea/test/common/assertions/Optionals.java

```diff
@@ -0,0 +1,32 @@
+package org.triplea.test.common.assertions;
+
+import java.util.Optional;
+
+import org.hamcrest.Matcher;
+import org.triplea.test.common.CustomMatcher;
+
+/**
+ * Utility class with hamcrest matchers for {@code java.util.Optional}.
+ */
+public final class Optionals {
```

Comment:
Any objection to using a third-party library that provides equivalent functionality?  That seems better than maintaining our own code.

The [official Hamcrest issue](https://github.com/hamcrest/JavaHamcrest/issues/82) on this topic mentions two alternatives:

* https://github.com/npathai/hamcrest-optional
* https://github.com/spotify/java-hamcrest

They both offer the same functionality with respect to `Optional` matchers.

Reply:
No objection @ssoloff 
- npathai API is very similar, I'll replace our version with this.

- spotify matchers:
  - includes a lot more than we really need
  - the API for optionals does not read that well for me

Reply:
BTW, the Spotify matchers are published as separate artifacts, so you can get the `Optional` matchers independently of the others.